d2/src/main/java/com/linkedin/d2/balancer/util/FileSystemDirectory.java

```diff
@@ -44,39 +44,46 @@ public class FileSystemDirectory
 
   private final String _d2FsDirPath;
   private String _d2ServicePath;
+  private final String _fsFileExtension;
 
   public FileSystemDirectory(String d2FsDirPath, String d2ServicePath)
+  {
+    this(d2FsDirPath, d2ServicePath, FILE_STORE_EXTENSION);
+  }
+
+  public FileSystemDirectory(String d2FsDirPath, String d2ServicePath, String fsFileExtension)
   {
     _d2FsDirPath = d2FsDirPath;
     _d2ServicePath = d2ServicePath;
+    _fsFileExtension = fsFileExtension;
   }
 
   public List<String> getServiceNames()
   {
-    return getFileListWithoutExtension(getServiceDirectory(_d2FsDirPath, _d2ServicePath));
+    return getFileListWithoutExtension(getServiceDirectory(_d2FsDirPath, _d2ServicePath), _fsFileExtension);
   }
 
   public void removeAllServicesWithExcluded(Set<String> excludedServices)
   {
     List<String> serviceNames = getServiceNames();
     serviceNames.removeAll(excludedServices);
-    removeAllPropertiesFromDirectory(getServiceDirectory(_d2FsDirPath, _d2ServicePath), serviceNames);
+    removeAllPropertiesFromDirectory(getServiceDirectory(_d2FsDirPath, _d2ServicePath), serviceNames, _fsFileExtension);
   }
 
   public void removeAllClustersWithExcluded(Set<String> excludedClusters)
   {
     List<String> serviceNames = getClusterNames();
     serviceNames.removeAll(excludedClusters);
-    removeAllPropertiesFromDirectory(getServiceDirectory(_d2FsDirPath, _d2ServicePath), serviceNames);
+    removeAllPropertiesFromDirectory(getServiceDirectory(_d2FsDirPath, _d2ServicePath), serviceNames, _fsFileExtension);
   }
 
-  public static void removeAllPropertiesFromDirectory(String path, List<String> properties)
+  public static void removeAllPropertiesFromDirectory(String path, List<String> properties, String fileExtension)
```

Comment:
These are public methods, can we make backward compatible changes here to avoid bump minor version?

Reply:
In the newest commit I have added backward compatible signatures.